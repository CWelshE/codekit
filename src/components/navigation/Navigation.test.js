// Author: Cody Welsh
// License: GPL 3.0

import test from 'ava';
import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Nav, { navRoutes } from './Navigation';

configure({ adapter: new Adapter() });

test('has links container', (t) => {
  const wrapper = shallow(<Nav/>);
  const links = wrapper.find('div.Nav');
  t.is(links.is('.Nav'), true);
});

test('has same number of links as supplied routes', (t) => {
  const wrapper = shallow(<Nav/>);
  const linksLength = wrapper.find('Link').length;
  t.true(linksLength === (navRoutes.length));
});

