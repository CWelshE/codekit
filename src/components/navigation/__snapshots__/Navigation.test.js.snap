// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without errors 1`] = `
<ul
  className="css-1hv5ft3"
>
  <a
    aria-current={null}
    className="css-xd43de"
    href="/home"
    onClick={[Function]}
  />
  <div>
    <a
      aria-current={null}
      className="css-1iavrx4"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="css-1iavrx4"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      aria-current={null}
      className="css-1iavrx4"
      href="/users"
      onClick={[Function]}
    >
      Users
    </a>
  </div>
</ul>
`;
